README.md

```diff
@@ -1,13 +1,187 @@
+
+## Execution using the nextflow executable
+
+Executing a workflow  is a bit more involved and requires familo. 
+
+### Setup
+
+In order to use the workflows in this package on your local computer, you'll need to do the following:
+* Install [nextflow](https://www.nextflow.io/docs/latest/install.html)
+* Install a nextflow compatible executor. This workflow provides a profile for [docker](https://docs.docker.com/get-started/).
+
+### Location of the workflow scripts
+
+Nextflow workflow scripts, schema's and configuration files can be found in the `target/nextflow` folder.
+On the `main` branch however, only the source code that needs to be build into the functionning workflows and components can be found.
+Instead, please refer to the `main_build` branch or any of the tags to find the `target` folders.
+Components and workflows are organized into namespaces, which can be nested. Workflows are located at `target/nextflow/workflows`,
+while components that execute individual workflow steps are 
+
+A reference of workflows and modules is also provided in the [documentation](https://openpipelines.bio/components/).
+
+### Retreiving a list of a workflow parameters
+
+A list of workflows arguments can be consulted in multiple ways
+
+* On [Viash Hub](https://www.viash-hub.com/packages/openpipeline/latest) 
+* In the [reference documentation](https://openpipelines.bio/components/)
+* The config YAML file lists the argument for each workflow and component
+* In the `target/nextflow` folder, a nextflow schema JSON file (`nextflow_schema.json`) is provided next to each workflow `.nf` file.
+* Using nextflow on the CLI: 
+
+```bash
+nextflow run openpipelines-bio/openpipeline \
+    -r 2.1.1 \
+    -main-script target/nextflow/workflows/ingestion/demux/main.nf \
+    --help
+```
+
+### Resource usage tuning
+
+Nextflow's labels can be used to specify the amount of resources a process can use. This workflow uses the following labels for CPU, memory and disk:
+* `lowmem`, `lowmem`, `midmem`, `highmem`, `veryhighmem`
+* `lowcpu`, `lowcpu`, `midcpu`, `highcpu`, `veryhighcpu`
+* `lowdisk`, `middisk`, `highdisk`, `veryhighdisk`
+
+The defaults for these labels can be found at `src/workflows/utils/labels.config`. Nextflow checks that the specified resources for a process do not exceed what is available on the machine and will not start if it does. Create your own config file to tune the labels to your needs, for example:
+
+```
+// Resource labels
+withLabel: verylowcpu { cpus = 2 }
+withLabel: lowcpu { cpus = 8 }
+withLabel: midcpu { cpus = 16 }
+withLabel: highcpu { cpus = 16 }
+
+withLabel: verylowmem { memory = 4.GB }
+withLabel: lowmem { memory = 8.GB }
+withLabel: midmem { memory = 16.GB }
+withLabel: highmem { memory = 32.GB }
+```
+
+When starting nextflow using the CLI, you can use `-c` to provide the file to nextflow and overwrite the defaults.
```

Comment:
I wonder whether some of the (more technical) docs should be moved to the openpipelines website and maybe the usage example should be focused more on the analyses?

We can provide a section on additional information for advanced usage or something?

WDYT?

Reply:
Maybe, but I also think some of this information is not even technical, but essential to run OP on the CLI. For example: the default labels far exceed the resourses that are available on a generic PC.